app/Controller/StatsController.php

```diff
@@ -66,9 +66,12 @@ public function stats() {
 	}
 
 	protected function _getTimeFilter() {
-		$filter = $this->Incident->filterTimes[$this->request->query('filter')];
-		if (isset($filter)) {
-			return $filter;
+		$getFilter=$this->request->query('filter');
+		if(isset($getFilter)) {
+			$filter = $this->Incident->filterTimes[$getFilter];
+			if (isset($filter)) {
+				return $filter;
+			} 
```

Comment:
It should fallback to returing all as well here, otherwise the method does not return anything here...